File: 프로그래머스/0/120904. 숫자 찾기/숫자 찾기.js
function solution(num, k) {
    var answer = 0;

    var nums = num.toString();
    var arr = nums.split("");
    // num을 문자열로 변환한 다음 배열로 분할합니다.

    var k_string = k.toString();
    // k도 문자열로 변환합니다.

    for (let i = 0; i < arr.length; i++) {
        if (arr[i] === k_string) {
            answer = i + 1;
            break;
        } // arr의 i번째 요소가 k_string과 같으면
          // 인덱스 번호 + 1을 answer에 할당하고 반복문을 종료합니다.
        else if (arr[i] !== k_string) {
            answer = -1;
        } // arr의 모든 요소가 k_string과 같지 않으면
          // answer에 -1을 할당합니다.
    }
    return answer;
}

File: 프로그래머스/lv0/120904. 숫자 찾기/숫자 찾기.java
class Solution {
    public int solution(int num, int k) {
        int answer = 0;
        
        String nums = Integer.toString(num);
		String[] arr = nums.split("");
        // num을 문자열, 배열로 차례대로 변환한다
		String k_string = Integer.toString(k);
        // 문자열 비교를 위해 k도 문자열로 변환한다
		for(int i = 0; i < arr.length; i++) { 
			if(arr[i].equals(k_string)) { 
				answer = i + 1;
				break; 
			} // arr의 i번째 원소가 k_string과 동일하면 
              // 인덱스 번호를 answer에 넣고 break문을 통해 빠져나간다
			else if (!arr[i].equals(k_string)) {
				answer = -1;
			} // arr의 모든 원소가 k_string과 동일하지 않을 시
              // answer에 -1을 넣어준다
		}
        
        return answer;
    }
}